golang/testsuite/testsuite_impl/wait_for_endpoint_availability_test/wait_for_endpoint_availability_test_.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020 - present Kurtosis Technologies LLC.
+ * All Rights Reserved.
+ */
+
+package wait_for_endpoint_availability_test
+
+import (
+	"github.com/kurtosis-tech/kurtosis-client/golang/networks"
+	"github.com/kurtosis-tech/kurtosis-client/golang/services"
+	"github.com/kurtosis-tech/kurtosis-libs/golang/lib/testsuite"
+	"github.com/kurtosis-tech/kurtosis-libs/golang/testsuite/services_impl/datastore"
+	"github.com/palantir/stacktrace"
+	"github.com/sirupsen/logrus"
+)
+const (
+	datastoreServiceId services.ServiceID = "datastore"
+
+	healthCheckUrlSlug = "health"
+	healthyValue       = "healthy"
+
+	waitForStartupTimeBetweenPolls = 1
+	waitForStartupMaxPolls = 15
+	waitInitialDelaySeconds = 1
+)
+
+
+type WaitForEndpointAbailabilityTest struct {
+	datastoreImage string
+}
+
+func NewWaitForEnpointAvailabilityTest(datastoreImage string) *WaitForEndpointAbailabilityTest {
+	return &WaitForEndpointAbailabilityTest{datastoreImage: datastoreImage}
+}
+
+func (test WaitForEndpointAbailabilityTest) Configure(builder *testsuite.TestConfigurationBuilder) {
+	builder.WithSetupTimeoutSeconds(60).WithRunTimeoutSeconds(60)
+}
+
+func (test WaitForEndpointAbailabilityTest) Setup(networkCtx *networks.NetworkContext) (networks.Network, error) {
+	datastoreConfigFactory := datastore.NewDatastoreContainerConfigFactory(test.datastoreImage)
```

Comment:
I'd recommend leaving the `Setup` method blank, and pushing this inside `Run` (that way, you can use the same `datastoreConfigFactory`. It's not bad as it is; just might be a bit cleaner inside `Run`